src/riscv/lib/src/state_backend.rs

```diff
@@ -226,17 +227,17 @@ pub trait ManagerClone: ManagerBase {
 }
 
 /// Manager wrapper around `M` whose regions are immutable references to regions of `M`
-pub struct Ref<'backend, M>(std::marker::PhantomData<fn(&'backend M)>);
+pub struct Ref<'backend, M>(PhantomData<&'backend M>, Infallible);
```

Comment:
I'm not sure I understand what the `Infallible` does here.

Reply:
Sure, but it was already the case that we're never constructing `Ref`, right? This is just enforcing that constraint?